### PR TITLE
test: add tests for new DEX

### PR DESCRIPTION
**Pull request by kingcre**
## Description

The purpose of this PR is to increase test coverage.

## Tasks

- [x] x/amm/keeper
- [x] x/amm/types
- [x] x/exchange/keeper
- [x] x/exchange/types
- [x] x/amm invariant checks
- [x] x/exchange invariant checks

### CODE REVIEW

**Comment by crypin** [comment]
[this failed lint part](https://github.com/crescent-network/crescent/pull/167/files#diff-ec49f679a2621f9587dbc3d1536d6b09d4a8ea604347514ae02e0b51962a8ffcR188) If you put sanity check because there is no possibility of this part occurring, why don't you panic it so that you can check the occurrence with simulation, fuzzing, etc. before going to production?

File: x/amm/keeper/farming.go [hunk 1]
```diff
@@ -138,7 +138,13 @@ func (k Keeper) TerminateFarmingPlan(ctx sdk.Context, plan types.FarmingPlan) er
...
 	k.SetFarmingPlan(ctx, plan)
 	if plan.IsPrivate {
-		k.SetNumPrivateFarmingPlans(ctx, k.GetNumPrivateFarmingPlans(ctx)-1)
+		numPlans := k.GetNumPrivateFarmingPlans(ctx)
+		if numPlans > 1 {
+			k.SetNumPrivateFarmingPlans(ctx, k.GetNumPrivateFarmingPlans(ctx)-1)
+		} else {
+			k.DeleteNumPrivateFarmingPlans(ctx)
+		}
+	} else {
```
Comment (crypin) [inline x/amm/keeper/farming.go:147]: Unless this part of the implementation is missing, I think it can be erased

File: x/amm/keeper/farming.go [hunk 2]
```diff
@@ -138,7 +138,13 @@ func (k Keeper) TerminateFarmingPlan(ctx sdk.Context, plan types.FarmingPlan) er
 	plan.IsTerminated = true
 	k.SetFarmingPlan(ctx, plan)
 	if plan.IsPrivate {
-		k.SetNumPrivateFarmingPlans(ctx, k.GetNumPrivateFarmingPlans(ctx)-1)
+		numPlans := k.GetNumPrivateFarmingPlans(ctx)
+		if numPlans > 1 {
+			k.SetNumPrivateFarmingPlans(ctx, k.GetNumPrivateFarmingPlans(ctx)-1)
```
Comment (crypin) [inline x/amm/keeper/farming.go:143]: It could be ```k.SetNumPrivateFarmingPlans(ctx, numPlans-1)```

**Comment by kingcre** [comment]
> [this failed lint part](https://github.com/crescent-network/crescent/pull/167/files#diff-ec49f679a2621f9587dbc3d1536d6b09d4a8ea604347514ae02e0b51962a8ffcR188) If you put sanity check because there is no possibility of this part occurring, why don't you panic it so that you can check the occurrence with simulation, fuzzing, etc. before going to production?

This is being discussed separately. Another PR will resolve this issue later.

File: x/amm/keeper/farming.go (hunk 1 above)
Comment (kingcre) [reply x/amm/keeper/farming.go:147]: Thank you for finding it out!

**Review by crypin — APPROVED** [review]

**Pull request merged**